client.go

```diff
@@ -181,11 +181,28 @@ func newClient(v4 bool, v6 bool) (*client, error) {
 			log.Printf("[ERR] mdns: Failed to bind to udp6 port: %v", err)
 		}
 	}
-
 	if mconn4 == nil && mconn6 == nil {
 		return nil, fmt.Errorf("failed to bind to any multicast udp port")
 	}
 
+	// Check that unicast and multicast connections have been made for IPv4 and IPv6
+	// and disable the respective protocol if not.
+	if uconn4 == nil || mconn4 == nil {
+		log.Printf("[ERR] mdns: Failed to listen to both unicast and multicast on IPv4")
```

Comment:
There was a new logger field added in #116. Also as a nit, I would probably make this message `INFO` level because for some environments it might be normal not to setup IPV4 or IPV6 (there is also no precedent for `WARN` in this repo). 
```suggestion
		logger.Printf("[INFO] mdns: Failed to listen to both unicast and multicast on IPv4")
```